src/Scientist/Internals/Experiment.cs

```diff
@@ -79,8 +80,11 @@ public void Try(string name, Func<T> candidate)
         public void Ignore(Func<T, T, Task<bool>> block) => 
             _ignores.Add(block);
 
+        public void Context(string key, dynamic data) =>
+            _contexts.Add(key, data);
```

Comment:
This could throw if the key matches an existing key. Maybe we should follow the .NET convention and add a `bool TryAddContext(key, data)` method.